src/config.rs

```
@@ -51,3 +51,32 @@ pub fn is_lombok_enabled(configuration: &Option<Value>) -> bool {
         })
         .unwrap_or(true)
 }
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum UpdateCheckMode {
+    Always,
+    Once,
+    Never,
+}
+
+impl Default for UpdateCheckMode {
+    fn default() -> Self {
+        UpdateCheckMode::Always
+    }
+}
+
+pub fn get_update_check_mode(configuration: &Option<Value>) -> UpdateCheckMode {
+    if let Some(configuration) = configuration
+        && let Some(mode_str) = configuration
+            .pointer("/update_check_mode")
+            .and_then(|x| x.as_str())
+    {
+        return match mode_str.to_lowercase().as_str() {
+            "once" => UpdateCheckMode::Once,
+            "never" => UpdateCheckMode::Never,
+            "always" => UpdateCheckMode::Always,
+            _ => UpdateCheckMode::default(),
```

I wonder if supplying a malformed configuration should silently fall back to the default. What if the user set `"update_check_mode" : "no"`? That should be an error imo, or at least a warning (but we don't have a good way of communicating warnings as far as i know)

---

Unless you start zed from the CLI enabling foreground the communication of errors is very limited